### PR TITLE
Typecast Transformer

### PR DESCRIPTION
**Pull request by wwkeyboard**
Adds a typecasting transformer that takes a column name and a target datatype. It attempts to do the casting via typecasting functions added to the message View.

Still outstanding:
- [ ] Cast to string

### CODE REVIEW

File: myceliald/src/constructors/typecast_transformer.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use pipe::{config::Map, types::DynSection};
+use section::{command_channel::SectionChannel, message::DataType, SectionError};
+
+pub fn transformer<S: SectionChannel>(
+    config: &Map,
+) -> Result<Box<dyn DynSection<S>>, SectionError> {
+    let column = config
+        .get("column")
+        .ok_or("tagging requires a 'column' name")?
```
Comment (alexandrinaw) [inline myceliald/src/constructors/typecast_transformer.rs:9]: nit: Some of these lines still say "tagging" but other spots have been updated to say "typecasting"

File: myceliald/src/constructors/typecast_transformer.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        .as_str()
+        .ok_or("tagging 'column' name must be a string")?;
+    let target_type = match config
+        .get("target_type")
+        .ok_or("typecasting requires a 'target_type'")?
+        .as_str()
+        .ok_or("tagging 'text' must be set")?
```
Comment (alexandrinaw) [inline myceliald/src/constructors/typecast_transformer.rs:16]: nit: here the field name says `text` but should be `target_type`

**Review by alexandrinaw — COMMENTED** [review]
here's a diff you can apply to make the frontend work correctly with this new section: 
```diff
diff --git a/console/src/actions/clients.ts b/console/src/actions/clients.ts
index 0ab5372..5e80c6f 100644
--- a/console/src/actions/clients.ts
+++ b/console/src/actions/clients.ts
@@ -30,10 +30,10 @@ const formatSections = (client: clientFormatType, sections: any[], sources = tru
     }
     // FIXME: this section is called a source on the backend, so to have it appear in the UI correctly, we
     // have to tag it as both a source and a destination here
-    if (section.type === "tagging_transformer") {
+    if (section.type.endsWith("transformer")) {
       formatted.source = true;
       formatted.destination = true;
-      formatted.name = "tagging_transformer";
+      formatted.name = section.type;
     }
     return formatted;
   });
```

File: pipe/section/section_impls/typecast_transformer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+section = { path = "../../" }
+arrow = { version = "49", features = ["prettyprint"] }
+arrow_msg = { path = "../../../arrow_msg" }
+futures = "0.3"
```
Comment (dmzmk) [inline pipe/section/section_impls/typecast_transformer/Cargo.toml:12]: section re-exports futures

File: pipe/section/section_impls/typecast_transformer/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "typecast_transformer"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+section = { path = "../../" }
+arrow = { version = "49", features = ["prettyprint"] }
```
Comment (dmzmk) [inline pipe/section/section_impls/typecast_transformer/Cargo.toml:10]: Feels like we need to re-export arrow through arrow_msg, WDYT?

File: pipe/section/src/message.rs [hunk 5]
```diff
@@ -160,6 +161,161 @@ impl From<Vec<u8>> for Value {
...
+        match self {
+            Value::I8(v) => Ok(Value::Decimal(v.into())),
+            Value::I16(v) => Ok(Value::Decimal(v.into())),
+            Value::I32(v) => Ok(Value::Decimal(v.into())),
+            Value::I64(v) => Ok(Value::Decimal(v.into())),
+            Value::U8(v) => Ok(Value::Decimal(v.into())),
+            Value::U16(v) => Ok(Value::Decimal(v.into())),
+            Value::U32(v) => Ok(Value::Decimal(v.into())),
+            Value::U64(v) => Ok(Value::Decimal(v.into())),
+            // TODO: figure out how to cast these
```
Comment (dmzmk) [inline pipe/section/src/message.rs:309]: not sure what is the problem here exactly, rust_decimal provides methods to build Decimal from [f32/f64](https://docs.rs/rust_decimal/latest/rust_decimal/prelude/trait.FromPrimitive.html#method.from_f32)

**Review by dmzmk — APPROVED** [review]
lgtm!

File: myceliald/src/constructors/typecast_transformer.rs (hunk 1 above)
Comment (wwkeyboard) [reply myceliald/src/constructors/typecast_transformer.rs]: Thank you!

File: myceliald/src/constructors/typecast_transformer.rs (hunk 2 above)
Comment (wwkeyboard) [reply myceliald/src/constructors/typecast_transformer.rs]: Thank you!

File: pipe/section/section_impls/typecast_transformer/Cargo.toml (hunk 4 above)
Comment (wwkeyboard) [reply pipe/section/section_impls/typecast_transformer/Cargo.toml:10]: I think that would help with keeping the versions consistent.

File: pipe/section/section_impls/typecast_transformer/Cargo.toml (hunk 3 above)
Comment (wwkeyboard) [reply pipe/section/section_impls/typecast_transformer/Cargo.toml:12]: I'll remove this in the toString PR

File: pipe/section/src/message.rs (hunk 5 above)
Comment (wwkeyboard) [reply pipe/section/src/message.rs:309]: So we can return typecasterror from this function, but in the case of f32 and f64 they are only unrepresentable some times. You could have a value that works fine in development but rolls over at some point when it's out in the field. We need a better way to handle errors to make this friendly to use.

**Pull request merged**